### DISCUSSION TITLE
Some questions about kinematic orbit determination of LEO satellites

### DISCUSSION THREAD

**Discussion by LNTU-YS** [groops-devs/groops#251] [General]
Hello,

I'm trying to learn kinematic orbit determination of LEO satellites, but we encountered some problems.

This GNSS receiver of LEO satellite can receive carrier phases of L2L and L2W, but they don't exist at the same time. The location of the problem is shown in Figure 1. I tried to make it skip this part, but there was an error in "Estimate" as shown in Figure 2. How should I solve this problem?
![609c24c821097f1b4199ab8dada682b](https://github.com/groops-devs/groops/assets/111558200/c71d8c55-5965-4897-b7ff-036916826edf)
![微信图片_20240405205139](https://github.com/groops-devs/groops/assets/111558200/e1c90907-828b-4b99-ad1c-9b4842682037)

Another question is about “ResolveAmbiguities”. Imitating the example [scenario Leo Kinematic Orbit] you provided, I set the "ResolveAmbiguities" exactly the same. But after "ResolveAmbiguities", I got a worse result. So should I modify "sigmaMaxResolve"、"huber"or"huberPower" to get the best result?

Thank you in advance for your feedback and help.
With Kind Regards
YUAN

**Comment by zhedumi** [comment]
HI Yuan,

I do not think the issue is within the cycleslip repair since the error says it is already in the gnssProcessingStep and the cycleslip repair is usually done at the preprocessing stage. 
Would it be possible to see the whole log file and or scenario you created I can see more as to why the processing failed? I presume what you mean with L2W and L2L not being active at the same time means that it is either one or the other correct? 

Cheers 
Patrick

**Reply by LNTU-YS** [reply]
Hi Patrick

I have received the solution to the first question, you can read Torsten's answer.

Cheers
YUAN

**Comment by tmayerguerr** [comment]
Hi Yuan,

In the case of the raw observation approach, it is not easy to estimate the phase biases and integer ambiguities together. To minimize the number of ambiguities, we try to fix ambiguities at the same frequency in advance. Unfortunately, the algorithm does not work in your case, but I have not yet found an improved algorithm.

In order to be able to process sentinel 6a data anyway, I have installed a hack in which I rename all L2LG to L2WG:
```
diff --git a/source/gnss/gnssReceiverGenerator/gnssReceiverGeneratorLowEarthOrbiter.cpp b/source/gnss/gnssReceiverGenerator/gnssReceiverGeneratorLowEarthOrbiter.cpp
index 1902d803a..4442e5eb8 100644
--- a/source/gnss/gnssReceiverGenerator/gnssReceiverGeneratorLowEarthOrbiter.cpp
+++ b/source/gnss/gnssReceiverGenerator/gnssReceiverGeneratorLowEarthOrbiter.cpp
@@ -170,6 +170,21 @@ void GnssReceiverGeneratorLowEarthOrbiter::init(const std::vector<Time> &times,
           auto rotationCrf2Trf = std::bind(&EarthRotation::rotaryMatrix, earthRotation, std::placeholders::_1);
           recv->readObservations(fileNameObs, transmitters,  rotationCrf2Trf, timeMargin, elevationCutOff,
                                  useType, ignoreType, GnssObservation::RANGE | GnssObservation::PHASE);
+          // =================================
+          if(recv->name() == "sentinel6a") //HACK: replace L2LG -> L2WG
+          {
+            logWarning<<recv->name()<<": replace L2LG -> L2WG"<<Log::endl;
+            for(UInt idEpoch=0; idEpoch<recv->idEpochSize(); idEpoch++)
+              for(UInt idTrans=0; idTrans<recv->idTransmitterSize(idEpoch); idTrans++)
+              {
+                GnssObservation *obs = recv->observation(idTrans, idEpoch);
+                if(obs)
+                  for(UInt idType=0; idType<obs->size(); idType++)
+                    if(obs->at(idType).type == GnssType::L2_G+GnssType::L)
+                      obs->at(idType).type = GnssType::L2_G + GnssType::W + (obs->at(idType).type & GnssType::PRN);
+              }
+          }
+          // =================================
         }
       }
       catch(std::exception &e)

```

However, I hope to be able to solve the problem in general in the future, but for that I need a few days without the usual disruptions.

Regarding your second question:
Ambiguity resolution only works if the phase bias on the transmitter is determined in advance. This means that transmitter orbits, clocks and signal biases must match consistently.

best regards
Torsten

**Reply by LNTU-YS** [reply]
Hi Torsten

Is the term 'must match consistently' you mentioned referring to downloading transmitter orbits, clocks and signal biases from the same platform? I have already downloaded three types of data from GFZ before.  I referred to the settings of Sentinel-3 and obtained good results about ambiguity resolution, but Sentinel-6 cannot. So I'm confused.

best regards
YUAN
